ARCHITECTURE.md

```diff
@@ -0,0 +1,170 @@
+# kURL Architecture
+
+## Installer Manifest
+
+The kURL manifest is made up of top level add-ons, each with it's own properties for configuration.
+kURL.sh hosts an API where users can POST kURL manifests and a deterministic hash is stored and returned.
+This hashed can then be used to retrieve a kURL installation script or air-gap bundle.
+
+```yaml
+apiVersion: cluster.kurl.sh/v1beta1
+kind: Installer
+metadata:
+  name: 5e61e80
+spec:
+  kubernetes:
+    version: 1.23.5
+  containerd:
+    version: 1.4.13
+  weave:
+    version: 2.6.5
+  longhorn:
+    version: 1.2.2
+  ekco:
+    version: 0.19.0
+  minio:
+    version: 2020-01-25T02-50-51Z
+  contour:
+    version: 1.20.1
+  registry:
+    version: 2.7.1
+  kotsadm:
+    version: 1.69.1
+```
+
+## kURL URL
+
+A URL can be contructed including the hash to retrieve a kURL script (curl https://kurl.sh/5e61e80) or air-gap bundle (curl -LO https://kurl.sh/bundle/5e61e80.tar.gz).
+Additionally, a version of kURL can be pinned in that URL (https://kurl.sh/version/v2022.04.19/5e61e80).
+
+## kURL.sh API
+
+The kURL.sh API can be used to create installers and retrieve install scripts and air-gap bundles.
```

Comment:
an example screenshot will be helpful

Reply:
Is the diagram [here](https://github.com/replicatedhq/kURL/pull/2874/files#diff-8f6366fd8e7a5fa30f7f05879999195b7cf5fa1e3d157822eb37f0e91d226cfdR71) sufficient?